### PR TITLE
Add BGP peer group names option

### LINKED ISSUE #798: Knob for BGP peer group names in eos_designs
**Issue Type**

- [x] Template enhancement
- [ ] Role enhancement
- [ ] Documentation enhancement (avd.sh)

**Is your feature request related to a problem? Please describe.**

Provide a knob to allow the user to modify names leveraged in `eos_designs` for bgp peer group names:

-   IPv4_UNDERLAY_PEERS
-   MLAG_IPv4_UNDERLAY_PEER
-   EVPN_OVERLAY_PEERS

Reference issue: #792

### PR DESCRIPTION
**Pull request by melkypie**
## Change Summary

Adds the ability to configure your own names for BGP peer groups

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Documentation content changes
- [ ] Other (please describe):

## Related Issue(s)

Fixes #798 

## Component(s) name

`arista.avd.eos_designs`

## Checklist:

- [X] My code has been rebased from devel before I start
- [X] I have read the [**CONTRIBUTING**](https://www.avd.sh/docs/contributing/) document.
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly
- [X] All new and existing tests passed ([`pre-commit`](https://www.avd.sh/docs/installation/development/#python-virtual-environment), `make linting` and `make sanity-lint` **([This fails due to a recent commit](https://github.com/aristanetworks/ansible-avd/commit/820e5c628fb1629048035df5b0d238a56945c096#r48297750) and not because of this)**).

### CODE REVIEW

File: ansible_collections/arista/avd/molecule/evpn_underlay_ebgp_overlay_ebgp/inventory/group_vars/DC1_FABRIC.yml [hunk 1]
```diff
@@ -27,10 +27,12 @@ vxlan_vlan_aware_bundles: true
 # bgp peer groups passwords
 bgp_peer_groups:
   IPv4_UNDERLAY_PEERS:
+    name: UNDERLAY_PEERS
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/evpn_underlay_ebgp_overlay_ebgp/inventory/group_vars/DC1_FABRIC.yml]: Minor detail, but I think we should use "UNDERLAY-PEERS" in the tests. all other names we have are with "-" where variables are with "_". Same for the others.

**Review by ClausHolbechArista — APPROVED** [review]
Impressive contribution. This is very helpful and the implementation is respectful for existing deployments.
Only a small comment on molecule test, which is only my OCD kicking in :)

File: ansible_collections/arista/avd/molecule/evpn_underlay_ebgp_overlay_ebgp/inventory/group_vars/DC1_FABRIC.yml (hunk 1 above)
Comment (melkypie) [reply ansible_collections/arista/avd/molecule/evpn_underlay_ebgp_overlay_ebgp/inventory/group_vars/DC1_FABRIC.yml]: Okay, it's changed. Wanted to ask a question on whether or not is it best for me to add new commits with requested changes or just rebase requested changes into existing commits and then just force push the branch? Whatever is easier to review, doesn't give me any struggle to rebase it into nicer looking commits.
Also two more questions, I have a bunch of changes lined up for eos_designs, is it best if I create an issue before I put them into PRs and how do you also feel about taking over PRs that seem to be abandoned?

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/molecule/evpn_underlay_ebgp_overlay_ebgp/inventory/group_vars/DC1_FABRIC.yml]: Re. requested changes please just add new commits. We will squash merge if the history becomes too messy.
Re. please submit an issue. It is fine if you put related changed into the same issue. That will allow us to have a discussion before you spend too much time on a PR.
Re. Some of the abandoned PRs may not be relevant any longer, or we have replaced them and forgot to clean up. Please let us know which ones you are considering? (Comment with mention on the PRs thanks)

**Pull request merged**